Client/src/Components/Graph/Graph.tsx

```diff
@@ -31,19 +35,74 @@ export default function Graph(props: IProps) {
   const outerHeight = props && props.outerHeight || 500
   const width = outerWidth - margin.left - margin.right
   const height = outerHeight - margin.top - margin.bottom
+  const extremeBoundaries = props.extremeBoundaries
 
   const [isXLog, setXToggle] = useState(false)
   const [isYLog, setYToggle] = useState(false)
+  const [showSettings, setSettingsToggle] = useState(false)
+  const [xUpperBound, setXUpperBound] = useState(extremeBoundaries.maxX)
+  const [xLowerBound, setXLowerBound] = useState(extremeBoundaries.minX)
+  const [yUpperBound, setYUpperBound] = useState(extremeBoundaries.maxY)
+  const [yLowerBound, setYLowerBound] = useState(extremeBoundaries.minY)
+  // This loops through the selected arrays to find the min/max for both x and y.
 
   const handleXScaleClick = () => {
+    if (xLowerBound <= 0 && !isXLog) {
+      setXLowerBound(1)
+    }
     setXToggle(!isXLog)
   }
 
   const handleYScaleClick = () => {
+    if (yLowerBound <= 0 && !isYLog) {
+      setYLowerBound(1)
+    }
     setYToggle(!isYLog)
   }
 
-  let active = [null, null, null, null]
+  const handleSettingsClick = () => {
+    setSettingsToggle(!showSettings)
+  }
+
+  const handleXUpperBoundChange = (event) => {
+    const { value } = event.target
+    if ((value < extremeBoundaries.maxX)) {
+      setXUpperBound(Math.ceil(extremeBoundaries.maxX))
+    }
+    else {
+      setXUpperBound(value)
+    }
+  }
+  const handleXLowerBoundChange = (event) => {
+    const { value } = event.target
+    if ((value <= 0 && isXLog) || (value > extremeBoundaries.minX)) {
+      setXLowerBound(Math.floor(extremeBoundaries.minX))
+    }
+    else {
+      setXLowerBound(value)
+    }
+  }
+
+  const handleYUpperBoundChange = (event) => {
+    const { value } = event.target
+    if ((value < extremeBoundaries.maxY)) {
+      setYUpperBound(Math.ceil(extremeBoundaries.maxY))
+    }
+    else {
+      setYUpperBound(value)
```

Comment:
https://www.tutorialspoint.com/How-to-limit-the-number-of-characters-allowed-in-form-input-text-field#:~:text=To%20give%20a%20limit%20to%20the%20input%20field%2C%20use%20the,characters%2C%20use%20the%20maxlength%20attribute. try looking if setting a character limit on an input would be easier then manualy checking the boundary

Reply:
The limit is more in place so that the entire dataset is always within the boundaries specified. I don't know if setting a character limit will really help make it simpler.